fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirResourceGroup.java

```diff
@@ -159,6 +159,11 @@ public void addLogicalResources(List<IDatabaseObject> group, String prefix) {
                 .setTablespace(fhirTablespace)
                 .addPrivileges(resourceTablePrivileges)
                 .enableAccessControl(this.sessionVariable)
+                // Add indexes to avoid dead lock issue of derby
+                // Derby requires all columns used in where clause to be indexed, otherwise whole table lock will be
+                // used instead of row lock, which can cause dead lock issue frequently during concurrent accesses.
+                .addIndex(IDX + tableName + CURRENT_RESOURCE_ID, CURRENT_RESOURCE_ID)
```

Comment:
Should this be a conditional deployment of the indices?  
E.g. use a translator / adapter to identify the index as derby only. 

Also, for the query, what query failed?

Reply:
these are not for re-dev, just discussion here to get on same page.

Reply:
I thought about this(making the change for derby only) also, but I believe the added new indexes should be able to improve performance for db2 also, because these 2 columns are really heavily used in sql of almost all resource search scenarios, so hopefully we can benefit from indexing them... 

The dead-locked sql is "SELECT R.RESOURCE_ID, R.LOGICAL_RESOURCE_ID, R.VERSION_ID, R.LAST_UPDATED, R.IS_DELETED, R.DATA, LR.LOGICAL_ID FROM Observation_RESOURCES R, Observation_LOGICAL_RESOURCES LR WHERE LR.LOGICAL_ID = ? AND R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID"

Reply:
Mind updating 162, to say also improve db2.  It'll keep someone from coming in and changing removing if derby goes away.